test/features/when.js

```diff
@@ -13,3 +13,14 @@ test('mix.when()', t => {
 
     t.true(called);
 });
+
+test('it passes the mix instance to the callback', t => {
+    mix.when(true, _mix => t.deepEqual(mix, _mix));
+});
+
+test('it returns the mix instance', t => {
+    t.deepEqual(
+        mix,
+        mix.when(true, () => {})
+    );
+});
```

Comment:
Lets remove this last test. Because of the way components work I think it's unnecessary. It doesn't matter what register() returns as its ignored and all mix components work this way.

Reply:
No problem, I just made this change. FWIW, I copied that test more or less as-is from a similar test in `coffee.js`

https://github.com/JeffreyWay/laravel-mix/blob/f0ef8ab2370cb58fe656c9f285f8cb3144e38f40/test/features/coffee.js#L8-L10